README.md

```diff
@@ -62,147 +80,67 @@ Then call your helpers as usual:
-
-  # Set the namespace you want to use for your own components
-  namespace "AnotherCustom::Form"
+```ruby
+# app/components/form/text_field_component.rb
+class Form::TextFieldComponent < ViewComponent::Form::TextFieldComponent
+  def html_class
+    class_names("custom-text-field", "has-error": method_errors?)
+  end
 end
 ```
 
-Now let's generate your own components to customize their rendering. We can use the standard view_component generator:
-
-```console
-bin/rails generate component Custom::Form::TextField --inline --parent ViewComponent::Form::TextFieldComponent
-
-      invoke  test_unit
-      create  test/components/custom/form/text_field_component_test.rb
-      create  app/components/custom/form/text_field_component.rb
-```
-
-:warning: The `--parent` option is available since ViewComponent [`v2.41.0`](https://viewcomponent.org/CHANGELOG.html#2410). If you're using a previous version, you can always edit the generated `Custom::Form::CustomTextFieldComponent` class to make it inherit from `ViewComponent::Form::TextFieldComponent`.
+In this case we're leveraging the [`#class_names`](https://api.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-class_names) helper to:
 
-Change your forms to use your new builder:
+- always add the `custom-text-field` class;
+- add the `has-error` class if there is an error on the attribute (using `ViewComponent::Form::FieldComponent#method_errors?`).
 
-```diff
-- <%= form_for @user, builder: ViewComponent::Form::Builder do |f| %>
-+ <%= form_for @user, builder: CustomFormBuilder do |f| %>
-```
+### Adding your own custom helpers and components
 
-You can then customize the behavior of your `Custom::Form::CustomTextFieldComponent`:
+Add the helper method to your `ViewComponent::Form::Builder`
 
 ```rb
-# app/components/custom/form/text_field_component.rb
+# app/helpers/form_builder.rb
+class FormBuilder < ViewComponent::Form::Builder
+  def year_field(method, options = {})
+    render_component(:year_field, @object_name, method, objectify_options(options))
+  end
 
-class Admin::Form::TextFieldComponent < ViewComponent::Form::TextFieldComponent
-  def html_class
-    class_names("custom-text-field", "has-error": method_errors?)
+  def money_field(method, currencies = [], options = {})
+    render_component(:money_field, @object_name, method, currencies, objectify_options(options))
   end
 end
 ```
 
-In this case we leverage the [`#class_names`](https://api.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-class_names) helper to:
-- always add the `custom-text-field` class;
-- add the `has-error` class if there is an error on the attribute (using `ViewComponent::Form::FieldComponent#method_errors?`).
+Add your component which can optionally inherit from:
 
-The rendered form field will now look like this:
+- ViewComponent::Form::FieldComponent (suggested when adding a field because of helpers)
+- ViewComponent::Form::BaseComponent
```

Comment:
```suggestion
- `ViewComponent::Form::FieldComponent` (suggested when adding a field because of helpers)
- `ViewComponent::Form::BaseComponent`
```